src/chat/Database.cpp

```diff
@@ -0,0 +1,38 @@
+// allkeys.cpp
+// chris busch
+
+#include "Database.hpp"
+
+namespace chat {
+
+Record *Database::add_record() {
+    Record *record;
+    if ((record = new Record) == nullptr) {
+        return nullptr;
+    }
+    if (top_ != nullptr)
+        top_->next_ = record;
+    current_ = top_ = record;
+    top_->next_ = nullptr;
+    if (record_count_ == 0)
+        first_ = top_;
+    record_count_++;
+    return current_;
+}
+
+Record *Database::reset() { return (current_ = first_); }
```

Comment:
be aware the parens here are extraneous _and actually change the meaning in some (odd) contexts_. In general I'd avoid them. Specifically in `return` statements.

Also the assignment in a return itself is pretty uncommon and surprising. A linter may explode here as it looks like a comparison.

If nobody uses the return value, I'd drop it.

Reply:
I will check.  It came from the original.

Reply:
Yep I can remove the return.